osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -169,25 +172,32 @@ private double computeSpeedValue()
 
         private double computeAccuracyValue()
         {
-            // This percentage only considers HitCircles of any value - in this part of the calculation we focus on hitting the timing hit window
-            double betterAccuracyPercentage;
             int amountHitObjectsWithAccuracy = Attributes.HitCircleCount;
 
-            if (amountHitObjectsWithAccuracy > 0)
-                betterAccuracyPercentage = ((countGreat - (totalHits - amountHitObjectsWithAccuracy)) * 6 + countOk * 2 + countMeh) / (double)(amountHitObjectsWithAccuracy * 6);
-            else
-                betterAccuracyPercentage = 0;
+            if (amountHitObjectsWithAccuracy == 0)
+                return 0;
 
-            // It is possible to reach a negative accuracy with this formula. Cap it at zero - zero points
-            if (betterAccuracyPercentage < 0)
-                betterAccuracyPercentage = 0;
+            // This section should be documented by Tr3, but effectively we're calculating the exact same way as before, but
+            // we calculate a variance based on the object count and # of 50s, 100s, etc. This prevents us from having cases
+            // where an SS on lower OD is actually worth more than a 95% on OD11, even though the OD11 requires a greater
+            // window of precision.
 
-            // Lots of arbitrary values from testing.
-            // Considering to use derivation from perfect accuracy in a probabilistic manner - assume normal distribution
-            double accuracyValue = Math.Pow(1.52163, Attributes.OverallDifficulty) * Math.Pow(betterAccuracyPercentage, 24) * 2.83;
+            double p100 = (2 * (double)countOk) / amountHitObjectsWithAccuracy; // this is multiplied by two to encourage better accuracy. (scales better)
+            double p50 = (1 * (double)countMeh) / amountHitObjectsWithAccuracy;
+            double pm = (1 * (double)countMiss) / amountHitObjectsWithAccuracy;
+            double p300 = 1.0 - pm - p100 - p50;
 
-            // Bonus for many hitcircles - it's harder to keep good accuracy up for longer
-            accuracyValue *= Math.Min(1.15, Math.Pow(amountHitObjectsWithAccuracy / 1000.0, 0.3));
+            double m300 = 79.5 - 6.0 * Attributes.OverallDifficulty;
+            double m100 = 139.5 - 8.0 * Attributes.OverallDifficulty;
+            double m50 = 199.5 - 10.0 * Attributes.OverallDifficulty;
+
+            double variance = p300 * Math.Pow(m300 / 2.0, 2.0) +
+                              p100 * Math.Pow((m300 + m100) / 2.0, 2.0) +
+                              p50 * Math.Pow((m100 + m50) / 2.0, 2.0) +
+                              pm * Math.Pow(229.5 - 11 * Attributes.OverallDifficulty, 2.0);
+
+            double accuracyValue = 2.83 * Math.Pow(1.52163, (79.5 - 2 * Math.Sqrt(variance)) / 6.0)
+                                        * Math.Pow(Math.Log(1.0 + (Math.E - 1.0) * (Math.Min(amountHitObjectsWithAccuracy, 1600) / 1000.0)), 0.5);
```

Comment:
Something that catches my eye here: the argument of the natural log here is:

```
1.0 + (Math.E - 1.0) * (Math.Min(amountHitObjectsWithAccuracy, 1600) / 1000.0)
```

So it looks like logarithmic scaling to the [0,1] interval, but the second part in the multiplication isn't a [0,1] range value, it's a [0,1.6] range one. It seems that this should be divided by 1.6 to get the unit interval?

This may especially matter given that sqrt changes behaviour around the 1 argument (namely `sqrt(x) < x` for `x > 1`, and `sqrt(x) > x` for `x <= 1`).

Reply:
The cap at 1600 objects is to prevent infinite scaling for length, which is the same as how the old system handled this. If we wanted, we could change this to be a logarithmic scale after 1600 objects. Tr3's original alg did not have a cap here. I am fine either way.